lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -52,13 +57,31 @@ def get_collection_versions(self, requirement):
             if isinstance(requirement.src, GalaxyAPI)
             else self._apis
         )
-        return set(
-            (version, api)
-            for api in api_lookup_order
-            for version in api.get_collection_versions(
-                requirement.namespace, requirement.name,
-            )
-        )
+        result = set()
+        last_err = None
+        for api in api_lookup_order:
+            try:
+                for version in api.get_collection_versions(requirement.namespace, requirement.name):
```

Comment:
I feel like is becoming complicated. I suggest you explore refactoring a part of this into a helper generator only wrapping with try/except a specific invention as opposed to the block with a few levels of nesting. That is if you feel like exploring this, it's not a blocker.

Reply:
I think it's better now with f4df143. The cyclomatic complexity was 8, now it's 5.